db/migrate/20241030122707_create_note_tags.rb

```diff
@@ -0,0 +1,13 @@
+class CreateNoteTags < ActiveRecord::Migration[7.2]
+  def change
+    # Create a table and primary key
+    create_table :note_tags, :primary_key => [:note_id, :k] do |t|
+      t.column "note_id", :bigint, :null => false
+      t.column "k",  :string, :default => "", :null => false
+      t.column "v",  :string, :default => "", :null => false
+    end
+
+    # Add foreign key without validation
+    add_foreign_key :note_tags, :notes, :column => :note_id, :name => "note_tags_id_fkey", :validate => false
```

Comment:
Would moving this inside the `create_table` as `t.foreign_key` avoid the need to create it with validation disabled? Not sure if the strong migrations is clever enough to know that is safe?

If not then you can just add a second migration to enable validation as it's safe to do on an empty table.